src/lib/swapper/LifiSwapper/approvalNeeded/approvalNeeded.ts

```diff
@@ -25,7 +25,10 @@ export const approvalNeeded = async ({
     })
   }
 
-  const { assetReference: sellAssetErc20Address } = fromAssetId(sellAsset.assetId)
+  const { assetReference: sellAssetErc20Address, assetNamespace } = fromAssetId(sellAsset.assetId)
+
+  // native assets do not require approval
+  if (assetNamespace !== ASSET_NAMESPACE.erc20) return { approvalNeeded: false }
```

Comment:
We probably want to add `ASSET_NAMESPACE.bep20` here as well.

Alternatively, checking that the asset is a native asset, rather than the fact that the asset is not a token might be a safer way:
- the ZRX swapper has a `isNativeEvmAsset()` method we could redeclare in web, or
- `sellAsset.assetId === adapter.getFeeAssetId()` is a one-liner that will achieve the same thing

Reply:
nice, im a fan of the second option because its self-maintaining